lib/cli/watch-run.js

```diff
@@ -184,8 +213,17 @@ const createWatcher = (
     rerunner.run();
   });
 
-  watcher.on('all', () => {
-    rerunner.scheduleRun();
+  watcher.on('all', (event, filePath) => {
+    if (watchFilesFlat.includes(filePath)) {
+      rerunner.scheduleRun();
+    } else {
+      if (event === 'add') {
+        regenerateWatchFilesFlat();
+        if (watchFilesFlat.includes(filePath)) {
+          rerunner.scheduleRun();
+        }
+      }
```

Comment:
[Question] Is there a reason not to always regenerate if `event` is `'add`'? I.e.:

```js
watcher.on('all', (event, filePath) => {
  if (event === 'add') {
    regenerateWatchFilesFlat();
  }

  if (watchFilesFlat.includes(filePath)) {
    rerunner.scheduleRun();
  }
});
```

I think being able to deduplicate the code makes it a little clearer to read.